service/handlers.go

```diff
@@ -0,0 +1,228 @@
+package service
+
+import (
+	"fmt"
+	"github.com/YuvalSteiy/book_service/data_store"
+	"github.com/YuvalSteiy/book_service/models"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+const (
+	INIT_BOOKSTORE_FAIL = "Book Store Failed To Initialize"
+	INIT_USERDATA_FAIL  = "User Data Failed To Initialize"
+)
+
+func GetBookByID(c *gin.Context) {
+	err := updateUserData(c)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"Error": "Failed To Update User Data"})
+		return
+	}
+
+	db := data_store.NewBookStorer()
+	if db == nil {
```

Comment:
you can remove this check - you are not starting the service if it's `nil` (and in the other functions)